### PR TITLE
ci: Fix get_changed_files_output

### PR DESCRIPTION
**Pull request by MarcoFalke**
`FETCH_HEAD` will actually refer to `origin/master` (aka `origin/HEAD`), and not the current pull request.

### CODE REVIEW

**Comment by jonathanmetzman** [comment]
I think this might be failing because github's checkout action doesn't clone a repo like a normal user does.

**Comment by jonathanmetzman** [comment]
Really appreciate this effort!

**Comment by MarcoFalke** [comment]
Not sure how to debug this further. 

* The stderr and stdout are empty (Normally git would print an error).
* `git log -1 --format=%H HEAD` works fine in the `get_changed_files_output` context.
* `git log -1 --format=%H origin/HEAD`, too.
* Locally a call to `merge-base` with the two hashes passes with the same git version that CI uses here.

File: .github/workflows/project_tests.yml [hunk 1]
```diff
@@ -48,8 +48,8 @@ jobs:
           access_token: ${{ github.token }}
 
       - uses: actions/checkout@v2
-      - run: |  # Needed for git diff to work.
-          git fetch origin master --depth 1
+      - run: |  # Needed for git diff to work. (get_changed_files_output)
+          git fetch origin master --unshallow
```
Comment (MarcoFalke) [inline .github/workflows/project_tests.yml:52]: This uses an additional 40MB of disk space and 2 seconds wall clock time, which I believe is acceptable

**Comment by MarcoFalke** [comment]
Pushed a commit to also fix the presubmit checks

File: .github/workflows/project_tests.yml (hunk 1 above)
Comment (asraa) [reply .github/workflows/project_tests.yml:52]: can `fetch_depth 0` replace this? 
https://github.com/actions/checkout#fetch-all-history-for-all-tags-and-branches

**Review by asraa — APPROVED** [review]
LGTM!

**Comment by inferno-chromium** [comment]
Wow, thanks a lot!
@jonathanmetzman @oliverchang as fyi

**Pull request merged**